keps/sig-storage/695-skip-permission-change/README.md

```diff
@@ -0,0 +1,276 @@
+  observability if this feature?**
+  As documented above we already have metrics for tracking mount timing. We are planning to add a metric
+  for time it takes to change permission of a volume before mount but this is not necessary for observability of this feature.
+
+### Dependencies
+
+* **Does this feature depend on any specific services running in the cluster?**
+  Not applicable
+
+### Scalability
+
+* **Will enabling / using this feature result in any new API calls?**
+  This feature should cause no new API calls.
+
+* **Will enabling / using this feature result in introducing new API types?**
+  This feature introduces no new API types.
+
+* **Will enabling / using this feature result in any new calls to cloud
+  provider?**
+  No. This feature has no cloud-provider integration.
+
+* **Will enabling / using this feature result in increasing size or count
+  of the existing API objects?**
+  Since this feature adds a new field to pod's securitycontext, it will increase API size of
+  Pod object:
+  Describe them providing:
+  - API type(s): Pod
+  - Estimated increase in size: (e.g. new annotation of size 32B): 35B
+
+
+* **Will enabling / using this feature result in increasing time taken by any
+  operations covered by [existing SLIs/SLOs][]?**
+  No. If anything this feature will reduce time it takes for a Pod to start.
+
+* **Will enabling / using this feature result in non-negligible increase of
+  resource usage (CPU, RAM, disk, IO, ...) in any components?**
+  No. This feature should not cause any increase in memory or CPU usage of the affected component.
+
+### Troubleshooting
+
+Troubleshooting section serves the `Playbook` role as of now. We may consider
+splitting it into a dedicated `Playbook` document (potentially with some monitoring
+details). For now we leave it here though.
+
+* **How does this feature react if the API server and/or etcd is unavailable?**
+  Not applicable
+
+* **What are other known failure modes?**
+  - A brown field volume has correct top level permission but incorrect leaf level permission.
+    - Detection: If user has used `fsGroupChangePolicy` as `OnRootMismatch` in pod's security context and somehow top level directory has right permission but subdirectories don't then when pod attempts to read/write those subdirectories it could fail. So there are no metrics for this error but could result in pod not running correctl.
+    - Mitigations: This failure does not affect existing workloads and user can switch to using `Always` `fsGroupChangePolicy` in pod's security context.
+    - Diagnostics: Pod does not run correctly.
+    - Testing: We do not attempt to check for leaf level perissions if `fsGroupChangePolicy` is set to `OnRootMismatch`(which would defeat the purpose of this feature). But since user needs to opt-in for using this feature, they should be aware of side effects.
+
+* **What steps should be taken if SLOs are not being met to determine the problem?**
```

Comment:
How about some of these steps?

- Check the Pod spec if the new field is being used
- Check volume mount latency/error metrics
- Check kubelet logs

Reply:
hmm, I thought about this. But I think a high percentage of mount errors for example does not automatically translate into SLOs not being met for mount operation because of this change. It could be something unrelated in the cluster too. I could still document this metric as a possible investigation/troubleshooting starting point.

Reply:
filled out this section for using the steps.